api4/resolver_test.go

```diff
@@ -143,3 +143,85 @@ func TestGraphQLChannelsLeft(t *testing.T) {
 		assert.Len(t, q.ChannelsLeft, 0)
 	})
 }
+
+func TestGraphQLRolesLoader(t *testing.T) {
+	os.Setenv("MM_FEATUREFLAGS_GRAPHQL", "true")
+	defer os.Unsetenv("MM_FEATUREFLAGS_GRAPHQL")
+
+	th := Setup(t).InitBasic()
+	defer th.TearDown()
+
+	var q struct {
+		User struct {
+			ID    string `json:"id"`
+			Roles []struct {
+				ID   string `json:"id"`
+				Name string `json:"Name"`
+			} `json:"roles"`
+		} `json:"user"`
+		ChannelMembers []struct {
+			MsgCount float64 `json:"msgCount"`
+			Roles    []struct {
+				ID   string `json:"id"`
+				Name string `json:"Name"`
+			} `json:"roles"`
+		} `json:"channelMembers"`
+		TeamMembers []struct {
+			SchemeUser bool `json:"schemeUser"`
+			Roles      []struct {
+				ID   string `json:"id"`
+				Name string `json:"Name"`
+			} `json:"roles"`
+		}
+	}
+
+	input := graphQLInput{
+		OperationName: "channelMembers",
+		Query: `
+	query channelMembers {
+	user(id: "me") {
+		id
+		username
+		roles {
+			id
+			name
+		}
+	}
+	channelMembers(userId: "me") {
+		msgCount
+		roles {
+			id
+			name
+		}
+	}
+	teamMembers(userId: "me") {
+		schemeUser
+		roles {
+			id
+			name
+		}
+	}
+	}
+	`,
+	}
+
+	resp, err := th.MakeGraphQLRequest(&input)
+	require.NoError(t, err)
+	require.Len(t, resp.Errors, 0)
+	require.NoError(t, json.Unmarshal(resp.Data, &q))
+
+	assert.Len(t, q.User.Roles, 1)
```

Comment:
I think we shouldn't continue if we have let's say `0` elements in the slice.
```suggestion
	require.Len(t, q.User.Roles, 1)
```